app/api/dao/program.py

```diff
@@ -0,0 +1,248 @@
+            "program_name": program.program_name,
+            "organization_id": program.organization_id,
+            "organization_name": organization.name,
+            "representative_id": user["id"],
+            "representative_name": user["name"],
+            "start_date": program.start_date,  
+            "end_date": program.end_date,
+            "description": program.description,
+            "target_skills": program.target_skills,
+            "target_candidate_gender": program.target_candidate["target_candidate_gender"],
+            "target_candidate_age": program.target_candidate["target_candidate_age"],
+            "target_candidate_ethnicity": program.target_candidate["target_candidate_ethnicity"],
+            "target_candidate_sexual_orientation": program.target_candidate["target_candidate_sexual_orientation"],
+            "target_candidate_religion": program.target_candidate["target_candidate_religion"],
+            "target_candidate_physical_ability": program.target_candidate["target_candidate_physical_ability"],
+            "target_candidate_mental_ability": program.target_candidate["target_candidate_mental_ability"],
+            "target_candidate_socio_economic": program.target_candidate["target_candidate_socio_economic"],
+            "target_candidate_highest_education": program.target_candidate["target_candidate_highest_education"],
+            "target_candidate_years_of_experience": program.target_candidate["target_candidate_years_of_experience"],
+            "target_candidate_other": program.target_candidate["target_candidate_other"],
+            "payment_currency": program.payment_currency,
+            "payment_amount": program.payment_amount,
+            "contact_type": program.contact_type.value,
+            "zone": program.zone.value,
+            "student_responsibility": program.student_responsibility,
+            "mentor_responsibility": program.mentor_responsibility,
+            "organization_responsibility": program.organization_responsibility,
+            "student_requirements": program.student_requirements,
+            "mentor_requirements": program.mentor_requirements,
+            "resources_provided": program.resources_provided,
+            "contact_name": program.contact_name,
+            "contact_department": program.contact_department,
+            "program_address": program.program_address,
+            "contact_phone": program.contact_phone,
+            "contact_mobile": program.contact_mobile,
+            "contact_email": program.contact_email,
+            "program_website": program.program_website,
+            "irc_channel": program.irc_channel,
+            "tags": program.tags,
+            "status": program.status.value,
+            "creation_date":program.creation_date
+        }
+    except TypeError as e:
+        return e, HTTPStatus.BAD_REQUEST
+        
+
+def update(program, data, success_message, status_code):
+    target_candidate_data = {}
+    try:
```

Comment:
I think some info should be given if it is a bad request, such as which field(s) is malformed

Reply:
@ramitsawhney27 , I do it this way coz for the fields that required specific format, the validation already done specific to each fields in the preceding code (inside app/api/resources/organizations.py) line 105
<img width="919" alt="Screen Shot 2020-08-20 at 8 46 41 am" src="https://user-images.githubusercontent.com/29667122/90697549-fc703900-e2c1-11ea-93c7-3bae87aad723.png">

as for the TypeError `e` done here, will specify which fields other than those catch in the above that is malformed. Let me know if you have another alternative. Thanks